sale_order_type/models/sale.py

```diff
@@ -51,6 +51,10 @@ def onchange_type_id(self):
                 vals.update({"incoterm": order_type.incoterm_id})
             if vals:
                 order.update(vals)
+            if order_type.route_id:
+                order.order_line.write({"route_id": order_type.route_id.id})
+            else:
+                order.order_line.write({"route_id": False})
```

Comment:
```suggestion
            order.order_line.write({"route_id": order_type.route_id.id})
```

This is basically the same in only 1 line of code.
Can you check, though, about this errors that appears when changing the order type if the order is not saved yet?
I left a picture and video in one of the comments, I think it's still not resolved

Reply:
I checked the picture and video & tested multiple times, But I can't see the same error.
It's working properly.